cmd2/cmd2.py

```diff
@@ -381,6 +404,288 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, *,
+                            if attached_cmdset is not None and attached_cmdset is not cmdset:
+                                raise CommandSetRegistrationError(
+                                    'Cannot uninstall CommandSet when another CommandSet depends on it')
+                            check_parser_uninstallable(subparser)
+            if command_parser is not None:
+                check_parser_uninstallable(command_parser)
+
+    def _register_subcommands(self, cmdset: Union[CommandSet, 'Cmd']) -> None:
+        """
+        Register subcommands with their base command
+
+        :param cmdset: CommandSet or cmd2.Cmd subclass containing subcommands
+        """
+        if not (cmdset is self or cmdset in self._installed_command_sets):
+            raise CommandSetRegistrationError('Cannot register subcommands with an unregistered CommandSet')
+
+        # find all methods that start with the subcommand prefix
+        methods = inspect.getmembers(
+            cmdset,
+            predicate=lambda meth: isinstance(meth, Callable)
+            and hasattr(meth, constants.SUBCMD_ATTR_NAME)
+            and hasattr(meth, constants.SUBCMD_ATTR_COMMAND)
+            and hasattr(meth, constants.CMD_ATTR_ARGPARSER)
+        )
+
+        # iterate through all matching methods
+        for method_name, method in methods:
+            subcommand_name = getattr(method, constants.SUBCMD_ATTR_NAME)
+            full_command_name = getattr(method, constants.SUBCMD_ATTR_COMMAND)  # type: str
+            subcmd_parser = getattr(method, constants.CMD_ATTR_ARGPARSER)
+            parser_args = getattr(method, constants.SUBCMD_ATTR_PARSER_ARGS, {})
+
+            command_tokens = full_command_name.split()
+            command_name = command_tokens[0]
+            subcommand_names = command_tokens[1:]
+
+            # Search for the base command function and verify it has an argparser defined
+            if command_name in self.disabled_commands:
+                command_func = self.disabled_commands[command_name].command_function
+            else:
+                command_func = self.cmd_func(command_name)
+
+            if command_func is None:
+                raise CommandSetRegistrationError('Could not find command "{}" needed by subcommand: {}'
+                                                  .format(command_name, str(method)))
+            command_parser = getattr(command_func, constants.CMD_ATTR_ARGPARSER, None)
+            if command_parser is None:
+                raise CommandSetRegistrationError('Could not find argparser for command "{}" needed by subcommand: {}'
+                                                  .format(command_name, str(method)))
+
+            if isinstance(cmdset, CommandSet):
+                command_handler = _partial_passthru(method, self)
+            else:
+                command_handler = method
+            subcmd_parser.set_defaults(handler=command_handler)
```

Comment:
I think `handler` should be renamed to something more unique since it's possible for a user to already have an argument called handler.

Make this name a constant so users can just do: 
`func = getattr(ns, CONSTANT, None)`